src/lib/log.c

```diff
@@ -773,3 +773,90 @@ void xdebug_close_log()
 	fclose(XG_LIB(log_file));
 	XG_LIB(log_file) = NULL;
 }
+
+static int xdebug_has_named_mode(char *name, int len)
+{
+	if (strncmp(name, "off", len) == 0) {		
+		return XDEBUG_MODE_IS_OFF();		
+	}
+	if (strncmp(name, "develop", len) == 0) {
+		return XDEBUG_MODE_IS(XDEBUG_MODE_DEVELOP);
+	}
+	if (strncmp(name, "coverage", len) == 0) {		
+		return XDEBUG_MODE_IS(XDEBUG_MODE_COVERAGE);
+	}
+	if (strncmp(name, "debug", len) == 0) {
+		return XDEBUG_MODE_IS(XDEBUG_MODE_STEP_DEBUG);
+	}
+	if (strncmp(name, "gcstats", len) == 0) {
+		return XDEBUG_MODE_IS(XDEBUG_MODE_GCSTATS);
+	}
+	if (strncmp(name, "profile", len) == 0) {
+		return XDEBUG_MODE_IS(XDEBUG_MODE_PROFILING);
+	}
+	if (strncmp(name, "trace", len) == 0) {
+		return XDEBUG_MODE_IS(XDEBUG_MODE_TRACING);
+	}
+
+	return 0;
+}
+
+PHP_FUNCTION(xdebug_mode)
+{
+	char *mode = NULL;
+	size_t mode_len = 0;	
+	char *comma = NULL;
+	xdebug_str mode_out = XDEBUG_STR_INITIALIZER;
+	
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|s", &mode, &mode_len) == FAILURE) {
+		return;
+	}
+
+	if (mode) {
+		comma = strchr(mode, ',');
+		while (comma) {
+			if (!xdebug_has_named_mode(mode, comma - mode)) {
+				RETURN_FALSE;
+			}
+			mode = comma + 1;
+			while (*mode == ' ') {
+				mode++;
+			}
+			comma = strchr(mode, ',');
+		}
+		
+		if (!xdebug_has_named_mode(mode, strlen(mode))) {
+			RETURN_FALSE;
+		}
+
+		RETURN_TRUE;
+	}
+
+	if (XDEBUG_MODE_IS_OFF()) {
+		xdebug_str_add_literal(&mode_out, "off");
+	} else {
+		if (XDEBUG_MODE_IS(XDEBUG_MODE_DEVELOP)) {
+			xdebug_str_add_literal(&mode_out, "develop,");
```

Comment:
an array would be better

Reply:
I'm sure Derek is going to do his own thing here, but why do you think that an array would be better?

Reply:
well, to me, an array of strings seems easier to understand than a single string, where we need to remember that the string actually contains multiple modes in a comma-separated way (for which a check would probably be done by exploding the string anyway)

Reply:
@stof Yeah, I get that. But you don't need to explode the string to do an `in_array()` check, because you already have `xdebug_mode('modes,you,want,to,check')` functionality which returns true/false.

I have assumed that usage would be either to check a value (as above), or to set a value(s) in a new process - which you can do simply with `$mode = xdebug_mode().',modes,you,need'; etc`, rather than having to implode an array.

I also thought that if you returned an array of values, you should really only accept an arrary of values, for consistency, which seemed cumbersome if you only wanted to check a single value.

Reply:
`xdebug_mode().',modes,you,need'` is one example why you want array ;-)

Clean code is `array_unique([...xdebug_mode(), 'modes', 'you', 'need']))` :)

Reply:
@mvorisek Why? It doesn't matter (each value is just `OR`-ed to the current value) and is consistent with how the ini and env values are parsed/used.